.github/workflows/add_ported_warnings.yml

```diff
@@ -25,5 +25,30 @@ jobs:
         uses: Ana06/get-changed-files@v1.2
 
       - name: run the script
+        id: script
         run: |
           python scripts/detect_ported_files.py ${{ steps.changed-files.outputs.all }}
+
+      - id: PR
+        uses: 8BitJonny/gh-get-current-pr@2.2.0
+        # TODO: this may not work properly if the same commit is pushed to multiple branches:
+        # https://github.com/8BitJonny/gh-get-current-pr/issues/8
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          sha: ${{ github.event.pull_request.head.sha }}
+          # Only return if PR is still open
+          filterOutClosed: true
+
+      - if: steps.script.outputs.modifies_ported == 'True'
+        id: add_label
+        name: add "modifies-synchronized-file"
+        # we use curl rather than octokit/request-action so that the job won't fail
+        # (and send an annoying email) if the labels don't exist
+        run: |
+          curl -L \
+            -X POST \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}"\
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            https://api.github.com/repos/${{ github.repository }}/issues/${{ steps.PR.outputs.number }}/labels \
+            -d '{"labels":["modified-synchronized-file"]}'
```

Comment:
Does this clear all other labels?

Reply:
I dont think so, the difference is POST vs PUT https://docs.github.com/en/rest/issues/labels?apiVersion=2022-11-28#add-labels-to-an-issue